OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs

```diff
@@ -53,15 +54,11 @@ public void PlayMusic(string track = null, LuaFunction func = null)
 			if (!Game.Settings.Sound.MapMusic)
 				return;
 
-			var music = world.Map.Rules.InstalledMusic.Select(a => a.Value).ToArray();
-			if (!music.Any())
+			if (!playlist.IsInstalled())
 				return;
 
 			var musicInfo = !string.IsNullOrEmpty(track) ? world.Map.Rules.Music[track]
-			: Game.Settings.Sound.Repeat && previousMusic != null ? previousMusic
-			: Game.Settings.Sound.Shuffle ? music.Random(Game.CosmeticRandom)
-			: previousMusic == null ? music.First()
-			: music.SkipWhile(s => s != previousMusic).Skip(1).First();
+			: playlist.GetNextSong();
```

Comment:
Please throw an error if someone asks for a bogus track instead of silently falling back to a random song.